src/EditorFeatures/Core/AddImports/AbstractAddImportsPasteCommandHandler.cs

```diff
@@ -117,46 +125,49 @@ public void ExecuteCommand(PasteCommandArgs args, Action nextCommandHandler, Com
                 return;
             }
 
+            // We're showing our own UI, ensure the editor doesn't show anything itself.
+            executionContext.OperationContext.TakeOwnership();
+
+            var token = _listener.BeginAsyncOperation(nameof(ExecuteAsync));
+
+            ExecuteAsync(document, snapshotSpan, args.TextView)
+                .ReportNonFatalErrorAsync()
+                .CompletesAsyncOperation(token);
+        }
+
+        private async Task ExecuteAsync(Document document, SnapshotSpan snapshotSpan, ITextView textView)
+        {
```

Comment:
nit. consider adding: `_threadingContext.ThrowIfNotOnUIThread();`

Reply:
Interesting. Do we need this work to be on the UI thread to start?

Reply:
oh... will the indicator factory fail?

Reply:
It may be fine (I think I wrote things to be fine either way). However, this would be helpful for documenting how things bounce across threads. It's totally optional on your part of you want to do this here.